spread.yaml

```diff
@@ -417,6 +417,18 @@ backends:
             - opensuse-tumbleweed-64:
                   username: opensuse
                   password: opensuse
+    garden:
+        <<: *qemu
+        type: adhoc
+        allocate: |
+          # Spread does not expose any settings to ad-hoc backends.
+          export QEMU_MEM_OPTION="-m 4096"
+          export QEMU_SMP_OPTION="-smp $(nproc)"
+          # Convert snapd convention of system names to image-garden convention:
+          # - ubuntu-24.04-64 becomes ubuntu-cloud-24.04.$(uname -m)
+          # - ubuntu-core-24-64 becomes ubuntu-cloud-24.04.$(uname -m)
+          exec image-garden allocate "$(echo "$SPREAD_SYSTEM" | sed -e 's/\(\w\+\)/\1-cloud/' -e 's/\(\w\+\)-core-\(\w\+\)/\1-\2.04/'  -e 's/-64$/.'"$(uname -m)"'/g')"
```

Comment:
shouldn't this pass SPREAD_SYSTEM_USERNAME and SPREAD_SYSTEM_PASSWORD?

Reply:
That is interesting, I didn't know those are exposed. I can do that, yes. This will require changes upstream as the username/password are fixed at present.